ext/cdi/jersey-cdi1x/src/main/java/org/glassfish/jersey/ext/cdi1x/internal/AbstractCdiBeanSupplier.java

```diff
@@ -118,7 +119,8 @@ public void preDestroy(final T instance) {
      */
     private static <T> T produce(InjectionTarget<T> target, CreationalContext<T> ctx, InjectionManager im, Class<T> clazz) {
         try {
-            return target.produce(ctx);
+            return hasContextAnnotatedConstructors(clazz) ? im.create(clazz)
```

Comment:
We should also change what is inside the Exception block, that currently executes always `im.create(clazz);`, even if it was executed before.

But anyway, it is closed. We will discuss it next Monday probably.